controllers/certificatesigningrequests_controller.go

```diff
@@ -69,21 +71,40 @@ func NewCertificateSigningRequestsReconciler(mgr manager.Manager, clusterConfig
 // Reconcile is part of the main kubernetes reconciliation loop which reads that state of the cluster for a
 // CertificateSigningRequests object and aims to move the current state of the cluster closer to the desired state.
 func (r *certificateSigningRequestsReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = r.log.WithValues("certificatesigningrequests", req.NamespacedName)
+	log := r.log.WithValues("certificatesigningrequests", req.NamespacedName)
+
+	// Update operator condition to prevent OLM from updating WMCO while CSRs are being processed
+	if err := conditions.PatchUpgradeable(r.client, r.watchNamespace, meta.ConditionFalse,
+		conditions.UpgradeableFalseMessage, conditions.UpgradeableFalseReason); err != nil {
+		log.Info("unable to set Upgradeable condition, be cautious of automatic operator upgrades", "error", err)
+	}
 
 	certificateSigningRequest := &certificates.CertificateSigningRequest{}
 	if err := r.client.Get(ctx, req.NamespacedName, certificateSigningRequest); err != nil {
 		if k8sapierrors.IsNotFound(err) {
 			// Request object not found, could have been deleted after reconcile request.
 			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
 			// Return and don't requeue
+			// Communicate current operator condition Upgradeable=True to allow OLM to update WMCO if needed
+			if err := conditions.PatchUpgradeable(r.client, r.watchNamespace, meta.ConditionTrue,
+				conditions.UpgradeableTrueMessage, conditions.UpgradeableTrueReason); err != nil {
+				log.Info("unable to set Upgradeable condition, manual operator upgrades may be needed", "error", err)
+			}
```

Comment:
This can be a helper. Something like `condition.DisableOLMUpgrade()` and you wouldn't expose `PatchUpgradeable` and all the other constants.